packages/vow/src/rejection-tracker.js

```diff
@@ -25,73 +25,73 @@ const makeUnhandledRejection = (rejected, upgradedPK) => {
-    handle: M.call(VowCapShape).returns(),
-    reject: M.call(VowCapShape, M.raw(), M.promise()).returns(),
-  }),
-  upgradeRejectionWatcher: M.interface('UpgradeRejectionWatcher', {
-    onRejected: M.call(M.raw(), M.raw()).returns(),
-  }),
-};
-
 /** @param {Zone} zone */
 export const prepareVowRejectionTracker = zone => {
-  /** @type {WeakMap<VowPayloadV0, () => void>} */
+  /** @type {WeakMap<PassableCap, () => void>} */
   const vowToCancelUnhandledRejection = new WeakMap();
 
-  const makeVowRejectionTrackerKit = zone.exoClassKit(
-    'VowRejectionTrackerKit',
-    VowRejectionTrackerKitI,
-    () => ({}),
+  const upgradeRejectionWatcher = zone.exo(
+    'UpgradeRejectionWatcher',
+    M.interface('UpgradeRejectionWatcher', {
+      onRejected: M.call(M.raw(), M.raw()).returns(),
+    }),
     {
-      public: {
-        handle(vowV0) {
-          const cancel = vowToCancelUnhandledRejection.get(vowV0);
-          if (!cancel) {
-            throw Fail`vow ${vowV0} already handled`;
-          }
-          cancel();
-        },
-        /**
-         *
-         * @param {VowPayloadV0} vowV0
-         * @param {any} reason
-         * @param {Promise<void>} [rejected]
-         */
-        reject(vowV0, reason, rejected = Promise.reject(reason)) {
-          const { upgradeRejectionWatcher } = this.facets;
-          !vowToCancelUnhandledRejection.has(vowV0) ||
-            Fail`vow ${vowV0} already rejected`;
-
-          // Register a never-resolved native promise with liveslots, so it
-          // can reject on upgrade.
-          const upgradedPK = makePromiseKit();
-          zone.watchPromise(
-            upgradedPK.promise,
-            upgradeRejectionWatcher,
-            reason,
-          );
-
-          // Save the cancel function.
-          const cancel = makeUnhandledRejection(rejected, upgradedPK);
-          vowToCancelUnhandledRejection.set(vowV0, cancel);
-        },
-      },
-      upgradeRejectionWatcher: {
-        onRejected(upgradeReason, baseReason) {
-          if (baseReason instanceof Error) {
-            assert.note(baseReason, X`upgraded: ${upgradeReason}`);
-          }
-          throw baseReason;
-        },
+      onRejected(upgradeReason, baseReason) {
+        const reason = assert.error(
+          X`VOW_REJECTION ${baseReason} not handled before upgrade ${upgradeReason}`,
+        );
+        // console.warn(reason);
+        throw reason;
       },
     },
   );
 
-  const makeVowRejectionTracker = () => makeVowRejectionTrackerKit().public;
-  return makeVowRejectionTracker;
+  const vowRejectionTracker = harden({
+    /**
+     * @param {PassableCap} vowCap
+     */
+    handle(vowCap) {
+      const cancel = vowToCancelUnhandledRejection.get(vowCap);
+      if (!cancel) {
+        console.warn(
+          X`Now handling a VOW_REJECTION from a prior incarnation of ${vowCap}`,
+        );
+        return;
+      }
+      vowToCancelUnhandledRejection.set(vowCap, sink);
```

Comment:
I'm feeling a little unsure about keeping a recognizable entry for the vowCap here. It stresses the swingset gc for no good reason. From what I can tell, `handle` being called multiple times for the same vow should be abnormal. While we shouldn't error, I think it's acceptable to provide an erroneous diagnostic in that case (claiming we're handling a rejection from a previous incarnation).